presets/ragengine/main.py

```diff
@@ -93,14 +118,24 @@ def health_check():
     """,
 )
 async def index_documents(request: IndexRequest):
+    # Use a context manager for measuring the time
+    start_time = time.time()
     try:
         doc_ids = await rag_ops.index(request.index_name, request.documents)
         documents = [
             DocumentResponse(doc_id=doc_id, text=doc.text, metadata=doc.metadata)
             for doc_id, doc in zip(doc_ids, request.documents)
         ]
+        # Record successful request
+        rag_index_success.inc()
+        # Record latency
+        rag_index_latency.observe(time.time() - start_time)
```

Comment:
Can you add label for function invoking the metric - this will keep it easier to track/give some additional insight